### PR TITLE
Add Reusable Differential Drive Node

### LINKED ISSUE #60: Create ROS2 reusable nodes
Currently, there is a driver for each robot and coding something like that takes time. It could be improved by creating reusable nodes that can drive different parts of the robot, e.g. there could be a differential drive component that interacts with motors and encoders (Webots), and publishes Odometry and subscribes to Twist topic.

Here is the initial request:
https://github.com/cyberbotics/webots_ros2/pull/59#pullrequestreview-384859672

### PR DESCRIPTION
**Pull request by lukicdarkoo**
**Description**
Our objective is to create a reusable component that could be shared within all differential drive robots.

**Related Issues**
Closes #60 

**Affected Packages**
  - webots_ros2_epuck
  - webots_ros2_core
  - webots_ros2_tiago

**Tasks**
  - [x] Create differential drive node
  - [x] Integrate the node into e-puck robot
  - [x] Integrate the node into Tiago robot
  - [x] Create a new RViz config file specifically for Tiago

### CODE REVIEW

**Comment by DavidMansolino** [comment]
👀

File: webots_ros2_core/webots_ros2_core/differential_drive_node.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                 left_joint='left wheel motor',
+                 right_joint='right wheel motor',
+                 left_encoder='left wheel sensor',
+                 right_encoder='right wheel sensor',
+                 command_topic='/cmd_vel',
+                 odometry_topic='/odom',
+                 odometry_frame='odom',
+                 robot_base_frame='base_link'
+                 ):
+        super().__init__(name)
```
Comment (DavidMansolino) [inline webots_ros2_core/webots_ros2_core/differential_drive_node.py:39]:
```suggestion
        super().__init__(name, args)
```

File: webots_ros2_core/webots_ros2_core/differential_drive_node.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# Copyright 1996-2020 Cyberbotics Ltd.
```
Comment (DavidMansolino) [inline webots_ros2_core/webots_ros2_core/differential_drive_node.py:1]: It is probably worth mentioning this new class in this changelog: https://github.com/cyberbotics/webots_ros2/blob/0a291134bd626c49a1ded4911163ca3e8c5705f7/webots_ros2_core/CHANGELOG.rst

File: webots_ros2_tiago/launch/tiago.launch.py [hunk 3]
```diff
@@ -34,19 +34,33 @@ def generate_launch_description():
     webots = launch_ros.actions.Node(package='webots_ros2_core', node_executable='webots_launcher',
                                      arguments=arguments, output='screen')
     # Controller node
-    synchronization = launch.substitutions.LaunchConfiguration('synchronization', default=False)
+    synchronization = launch.substitutions.LaunchConfiguration(
+        'synchronization', default=False)
```
Comment (DavidMansolino) [inline webots_ros2_tiago/launch/tiago.launch.py:38]: We should probably update all the `test_flake8.py` test files to have a line length of 128 as you did here:
https://github.com/cyberbotics/epuck_ros2/blob/1d51f52f575a9cf57d2871aa7de5cf0012edc57f/epuck_ros2/test/test_flake8.py#L23

File: webots_ros2_tiago/launch/tiago.launch.py [hunk 4]
```diff
@@ -34,19 +34,33 @@ def generate_launch_description():
...
+        'webots_ros2_tiago'), 'resource', 'odometry.rviz')
+    rviz = launch_ros.actions.Node(package='rviz2', node_executable='rviz2', output='screen',
+                                   arguments=[
+                                       '--display-config=' + rviz_config],
+                                   condition=launch.conditions.IfCondition(use_rviz))
+
     return launch.LaunchDescription([
         webots,
         controller,
+        rviz,
```
Comment (DavidMansolino) [inline webots_ros2_tiago/launch/tiago.launch.py:57]: Rviz should be added to the dependencies: https://github.com/cyberbotics/webots_ros2/blob/0a291134bd626c49a1ded4911163ca3e8c5705f7/webots_ros2_tiago/package.xml#L17

File: webots_ros2_tiago/launch/tiago.launch.py (hunk 3 above)
Comment (lukicdarkoo) [reply webots_ros2_tiago/launch/tiago.launch.py:38]: That was my editor :/ Not sure why CI didn't raise an error, but I will increase

File: webots_ros2_core/CHANGELOG.rst [hunk 5]
```diff
@@ -8,6 +8,7 @@ Changelog for package webots_ros2_core
 * Added a new tf_publisher node to publish tf of desired Webots nodes.
 * Added the possibility to run nodes in synchronized mode (using the 'synchronization' parameter).
 * Added a laser_publisher module.
+* Added better support for differential drive robots ('WebotsDifferentialDriveNode')
```
Comment (DavidMansolino) [inline webots_ros2_core/CHANGELOG.rst:11]:
```suggestion
* Added better support for differential drive robots ('WebotsDifferentialDriveNode' class).
```

**Pull request merged**